

<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>PH525x series - Biomedical Data Science</title>
    
    <meta name="author" content="Rafael Irizarry and Michael Love">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <!-- Le HTML5 shim, for IE6-8 support of HTML elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Le styles -->
    <link href="http://genomicsclass.github.io/book/assets/themes/twitter/bootstrap/css/bootstrap.2.2.2.min.css" rel="stylesheet">
    <link href="http://genomicsclass.github.io/book/assets/themes/twitter/css/style.css?body=1" rel="stylesheet" type="text/css" media="all">
    <link href="http://genomicsclass.github.io/book/assets/themes/twitter/css/kbroman.css" rel="stylesheet" type="text/css" media="all">
    <link href="http://genomicsclass.github.io/book/assets/themes/twitter/css/knitr.css" rel="stylesheet" type="text/css" media="all">

    <!-- Le fav and touch icons -->
    <link href="http://genomicsclass.github.io/book/assets/themes/twitter/ico/favicon.ico" rel="icon" type="image/png">

    <!-- atom & rss feed -->
    <link href="http://genomicsclass.github.io/booknil" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="http://genomicsclass.github.io/booknil" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">

    <!-- knitr javascript -->
    <script src="http://genomicsclass.github.io/book/assets/themes/twitter/knitr.js" type="text/javascript"></script>

    <!-- mathjax javascript -->
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
      tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}
      });
    </script>

    <script src='https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>


  </head>

  <body>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container-narrow">
          <a class="brand" href="http://genomicsclass.github.io/book">PH525x series - Biomedical Data Science</a>
        </div>
      </div>
    </div>

    <div class="container-narrow">

      <div class="content">
        

<div class="page-header">
  <h2>PH525x series - Biomedical Data Science </h2>
</div>

<div class="row-fluid">
  <div class="span12">
    <h2 id="links-and-resources">Links and resources</h2>

<ul>
  <li><a href="http://github.com/genomicsclass/labs">R markdown source files</a></li>
  <li><a href="https://leanpub.com/dataanalysisforthelifesciences/">ePub version on Leanpub</a></li>
  <li><a href="pages/classes.html">Links to the HarvardX class pages</a></li>
  <li><a href="pages/resources.html">External resources and books</a></li>
  <li><a href="pages/more_help.html">Finding more help for data analysis</a></li>
</ul>

<h2 id="chapter-0---introduction">Chapter 0 - Introduction</h2>
<ul>
  <li><a href="pages/introduction.html">Introduction</a>  <a href="https://github.com/genomicsclass/labs/blob/master/intro/introduction.Rmd">[Rmd]</a></li>
  <li><a href="pages/getting_started.html">Getting started</a>  <a href="https://github.com/genomicsclass/labs/blob/master/intro/getting_started.Rmd">[Rmd]</a></li>
  <li><a href="pages/getting_started_exercises.html">Getting started exercises</a></li>
  <li><a href="pages/dplyr_intro.html">dplyr introduction</a>  <a href="https://github.com/genomicsclass/labs/blob/master/intro/dplyr_intro.Rmd">[Rmd]</a></li>
  <li><a href="pages/dplyr_intro_exercises.html">dplyr introduction exercises</a></li>
  <li><a href="pages/math_notation.html">Mathematical notation</a>  <a href="https://github.com/genomicsclass/labs/blob/master/intro/math_notation.Rmd">[Rmd]</a></li>
</ul>

<h2 id="chapter-1---inference">Chapter 1 - Inference</h2>
<ul>
  <li><a href="pages/random_variables.html">Random variables</a>  <a href="https://github.com/genomicsclass/labs/blob/master/inference/random_variables.Rmd">[Rmd]</a></li>
  <li><a href="pages/random_variables_exercises.html">Random variables exercises</a></li>
  <li><a href="pages/populations_and_samples.html">Populations and samples</a>  <a href="https://github.com/genomicsclass/labs/blob/master/inference/populations_and_samples.Rmd">[Rmd]</a></li>
  <li><a href="pages/populations_and_samples_exercises.html">Populations and samples exercises</a></li>
  <li><a href="pages/clt_and_t-distribution.html">CLT and t-distribution</a>  <a href="https://github.com/genomicsclass/labs/blob/master/inference/clt_and_t-distribution.Rmd">[Rmd]</a></li>
  <li><a href="pages/clt_and_t-distribution_exercises.html">CLT and t-distribution exercises</a></li>
  <li><a href="pages/clt_in_practice.html">CLT in practice</a>  <a href="https://github.com/genomicsclass/labs/blob/master/inference/clt_in_practice.Rmd">[Rmd]</a></li>
  <li><a href="pages/clt_in_practice_exercises.html">CLT in practice exercises</a></li>
  <li><a href="pages/t-tests_in_practice.html">t-test in practice</a>  <a href="https://github.com/genomicsclass/labs/blob/master/inference/t-tests_in_practice.Rmd">[Rmd]</a></li>
  <li><a href="pages/confidence_intervals.html">Confidence intervals</a>  <a href="https://github.com/genomicsclass/labs/blob/master/inference/confidence_intervals.Rmd">[Rmd]</a></li>
  <li><a href="pages/power_calculations.html">Power calculations</a>  <a href="https://github.com/genomicsclass/labs/blob/master/inference/power_calculations.Rmd">[Rmd]</a></li>
  <li><a href="pages/power_calculations_exercises.html">Power calculations exercises</a></li>
  <li><a href="pages/monte_carlo.html">Monte carlo</a>  <a href="https://github.com/genomicsclass/labs/blob/master/inference/monte_carlo.Rmd">[Rmd]</a></li>
  <li><a href="pages/monte_carlo_exercises.html">Monte carlo exercises</a></li>
  <li><a href="pages/permutation_tests.html">Permutation tests</a>  <a href="https://github.com/genomicsclass/labs/blob/master/inference/permutation_tests.Rmd">[Rmd]</a></li>
  <li><a href="pages/permutation_tests_exercises.html">Permutation tests exercises</a></li>
  <li><a href="pages/association_tests.html">Association tests</a>  <a href="https://github.com/genomicsclass/labs/blob/master/inference/association_tests.Rmd">[Rmd]</a></li>
  <li><a href="pages/association_tests_exercises.html">Association tests exercises</a></li>
</ul>

<h2 id="chapter-2---exploratory-data-analysis">Chapter 2 - Exploratory Data Analysis</h2>
<ul>
  <li><a href="pages/exploratory_data_analysis.html">Exploratory data analysis</a>  <a href="https://github.com/genomicsclass/labs/blob/master/eda/exploratory_data_analysis.Rmd">[Rmd]</a></li>
  <li><a href="pages/plots_to_avoid.html">Plots to avoid</a>  <a href="https://github.com/genomicsclass/labs/blob/master/eda/plots_to_avoid.Rmd">[Rmd]</a></li>
  <li><a href="pages/exploratory_data_analysis_exercises.html">Exploratory data analysis exercises</a></li>
</ul>

<h2 id="chapter-3---robust-statistics">Chapter 3 - Robust Statistics</h2>
<ul>
  <li><a href="pages/robust_summaries.html">Robust summaries</a>  <a href="https://github.com/genomicsclass/labs/blob/master/robust/robust_summaries.Rmd">[Rmd]</a></li>
  <li><a href="pages/ranktest.html">Rank tests</a>  <a href="https://github.com/genomicsclass/labs/blob/master/robust/ranktest.Rmd">[Rmd]</a></li>
  <li><a href="pages/robust_summaries_exercises.html">Robust summaries exercises</a></li>
</ul>

<h2 id="chapter-4---matrix-algebra">Chapter 4 - Matrix Algebra</h2>
<ul>
  <li><a href="pages/intro_using_regression.html">Introduction to using regression</a>  <a href="https://github.com/genomicsclass/labs/blob/master/matrixalg/intro_using_regression.Rmd">[Rmd]</a></li>
  <li><a href="pages/intro_using_regression_exercises.html">Introduction to using regression exercises</a></li>
  <li><a href="pages/matrix_notation.html">Matrix notation</a>  <a href="https://github.com/genomicsclass/labs/blob/master/matrixalg/matrix_notation.Rmd">[Rmd]</a></li>
  <li><a href="pages/matrix_notation_exercises.html">Matrix notation exercises</a></li>
  <li><a href="pages/matrix_operations.html">Matrix operations</a>  <a href="https://github.com/genomicsclass/labs/blob/master/matrixalg/matrix_operations.Rmd">[Rmd]</a></li>
  <li><a href="pages/matrix_operations_exercises.html">Matrix operations exercises</a></li>
  <li><a href="pages/matrix_algebra_examples.html">Matrix algebra examples</a>  <a href="https://github.com/genomicsclass/labs/blob/master/matrixalg/matrix_algebra_examples.Rmd">[Rmd]</a></li>
  <li><a href="pages/matrix_algebra_examples_exercises.html">Matrix algebra examples exercises</a></li>
</ul>

<h2 id="chapter-5---linear-models">Chapter 5 - Linear Models</h2>
<ul>
  <li><a href="pages/linear_models_intro.html">Linear models introduction</a>  <a href="https://github.com/genomicsclass/labs/blob/master/linear/linear_models_intro.Rmd">[Rmd]</a></li>
  <li><a href="pages/linear_models_intro_exercises.html">Linear models introduction exercises</a></li>
  <li><a href="pages/expressing_design_formula.html">Expressing design formula</a>  <a href="https://github.com/genomicsclass/labs/blob/master/linear/expressing_design_formula.Rmd">[Rmd]</a></li>
  <li><a href="pages/expressing_design_formula_exercises.html">Expressing design formula exercises</a></li>
  <li><a href="pages/linear_models_in_practice.html">Linear models in practice</a>  <a href="https://github.com/genomicsclass/labs/blob/master/linear/linear_models_in_practice.Rmd">[Rmd]</a></li>
  <li><a href="pages/linear_models_in_practice_exercises.html">Linear models in practice exercises</a></li>
  <li><a href="pages/standard_errors.html">Standard errors</a>  <a href="https://github.com/genomicsclass/labs/blob/master/linear/standard_errors.Rmd">[Rmd]</a></li>
  <li><a href="pages/standard_errors_exercises.html">Standard errors exercises</a></li>
  <li><a href="pages/interactions_and_contrasts.html">Interactions and contrasts</a>  <a href="https://github.com/genomicsclass/labs/blob/master/linear/interactions_and_contrasts.Rmd">[Rmd]</a></li>
  <li><a href="pages/interactions_and_contrasts_exercises.html">Interactions and contrasts exercises</a></li>
  <li><a href="pages/collinearity.html">Collinearity</a>  <a href="https://github.com/genomicsclass/labs/blob/master/linear/collinearity.Rmd">[Rmd]</a></li>
  <li><a href="pages/collinearity_exercises.html">Collinearity exercises</a></li>
  <li><a href="pages/qr_and_regression.html">QR and regression</a>  <a href="https://github.com/genomicsclass/labs/blob/master/linear/qr_and_regression.Rmd">[Rmd]</a></li>
  <li><a href="pages/linear_models_going_further.html">Linear models going further</a>  <a href="https://github.com/genomicsclass/labs/blob/master/linear/linear_models_going_further.Rmd">[Rmd]</a></li>
</ul>

<h2 id="chapter-6---inference-for-high-dimensional-data">Chapter 6 - Inference for High-Dimensional Data</h2>
<ul>
  <li><a href="pages/intro_to_highthroughput_data.html">Introduction to high-throughput data</a>  <a href="https://github.com/genomicsclass/labs/blob/master/advinference/intro_to_highthroughput_data.Rmd">[Rmd]</a></li>
  <li><a href="pages/intro_to_highthroughput_data_exercises.html">Introduction to high-throughput data exercises</a></li>
  <li><a href="pages/inference_for_highthroughput.html">Inference for high-throughput data</a>  <a href="https://github.com/genomicsclass/labs/blob/master/advinference/inference_for_highthroughput.Rmd">[Rmd]</a></li>
  <li><a href="pages/inference_for_highthroughput_exercises.html">Inference for high-throughput data exercises</a></li>
  <li><a href="pages/multiple_testing.html">Multiple testing</a>  <a href="https://github.com/genomicsclass/labs/blob/master/advinference/multiple_testing.Rmd">[Rmd]</a></li>
  <li><a href="pages/multiple_testing_exercises.html">Multiple testing exercises</a></li>
  <li><a href="pages/eda_for_highthroughput.html">EDA for high-throughput data</a>  <a href="https://github.com/genomicsclass/labs/blob/master/advinference/eda_for_highthroughput.Rmd">[Rmd]</a></li>
  <li><a href="pages/eda_for_highthroughput_exercises.html">EDA for high-throughput data exercises</a></li>
</ul>

<h2 id="chapter-7---statistical-modeling">Chapter 7 - Statistical Modeling</h2>
<ul>
  <li><a href="pages/modeling.html">Modeling</a>  <a href="https://github.com/genomicsclass/labs/blob/master/modeling/modeling.Rmd">[Rmd]</a></li>
  <li><a href="pages/modeling_exercises.html">Modeling exercises</a></li>
  <li><a href="pages/bayes.html">Bayes theorem</a>  <a href="https://github.com/genomicsclass/labs/blob/master/modeling/bayes.Rmd">[Rmd]</a></li>
  <li><a href="pages/bayes_exercises.html">Bayes theorem exercises</a></li>
  <li><a href="pages/hierarchical_models.html">Hierarchical models</a>  <a href="https://github.com/genomicsclass/labs/blob/master/modeling/hierarchical_models.Rmd">[Rmd]</a></li>
  <li><a href="pages/hierarchical_models_exercises.html">Hierarchical models exercises</a></li>
</ul>

<h2 id="chapter-8---distance-and-dimension-reduction">Chapter 8 - Distance and Dimension Reduction</h2>
<ul>
  <li><a href="pages/distance.html">Distance</a>  <a href="https://github.com/genomicsclass/labs/blob/master/highdim/distance.Rmd">[Rmd]</a></li>
  <li><a href="pages/distance_exercises.html">Distance exercises</a></li>
  <li><a href="pages/pca_motivation.html">PCA motivation</a>  <a href="https://github.com/genomicsclass/labs/blob/master/highdim/pca_motivation.Rmd">[Rmd]</a></li>
  <li><a href="pages/svd.html">SVD</a>  <a href="https://github.com/genomicsclass/labs/blob/master/highdim/svd.Rmd">[Rmd]</a></li>
  <li><a href="pages/svd_exercises.html">SVD exercises</a></li>
  <li><a href="pages/projections.html">Projections</a>  <a href="https://github.com/genomicsclass/labs/blob/master/highdim/projections.Rmd">[Rmd]</a></li>
  <li><a href="pages/rotations.html">Rotations</a>  <a href="https://github.com/genomicsclass/labs/blob/master/highdim/rotations.Rmd">[Rmd]</a></li>
  <li><a href="pages/mds.html">MDS</a>  <a href="https://github.com/genomicsclass/labs/blob/master/highdim/mds.Rmd">[Rmd]</a></li>
  <li><a href="pages/mds_exercises.html">MDS exercises</a></li>
  <li><a href="pages/PCA.html">PCA</a>  <a href="https://github.com/genomicsclass/labs/blob/master/highdim/PCA.Rmd">[Rmd]</a></li>
  <li><a href="pages/pca_svd.html">PCA and SVD in R</a>  <a href="https://github.com/genomicsclass/labs/blob/master/advinference/storage/pca_svd.Rmd">[Rmd]</a></li>
</ul>

<h2 id="chapter-9---practical-machine-learning">Chapter 9 - Practical Machine Learning</h2>
<ul>
  <li><a href="pages/clustering_and_heatmaps.html">Clustering and heatmaps</a>  <a href="https://github.com/genomicsclass/labs/blob/master/ml/clustering_and_heatmaps.Rmd">[Rmd]</a></li>
  <li><a href="pages/clustering_and_heatmaps_exercises.html">Clustering and heatmaps exercises</a></li>
  <li><a href="pages/conditional_expectation.html">Conditional expectation</a>  <a href="https://github.com/genomicsclass/labs/blob/master/ml/conditional_expectation.Rmd">[Rmd]</a></li>
  <li><a href="pages/conditional_expectation_exercises.html">Conditional expectation exercises</a></li>
  <li><a href="pages/smoothing.html">Smoothing</a>  <a href="https://github.com/genomicsclass/labs/blob/master/ml/smoothing.Rmd">[Rmd]</a></li>
  <li><a href="pages/smoothing_exercises.html">Smoothing exercises</a></li>
  <li><a href="pages/machine_learning.html">Machine learning</a>  <a href="https://github.com/genomicsclass/labs/blob/master/ml/machine_learning.Rmd">[Rmd]</a></li>
  <li><a href="pages/crossvalidation.html">Crossvalidation</a>  <a href="https://github.com/genomicsclass/labs/blob/master/ml/crossvalidation.Rmd">[Rmd]</a></li>
  <li><a href="pages/crossvalidation_exercises.html">Crossvalidation exercises</a></li>
</ul>

<h2 id="chapter-10---batch-effects">Chapter 10 - Batch Effects</h2>
<ul>
  <li><a href="pages/intro_to_batch_effects.html">Introduction to batch effects</a>  <a href="https://github.com/genomicsclass/labs/blob/master/batch/intro_to_batch_effects.Rmd">[Rmd]</a></li>
  <li><a href="pages/confounding.html">Confounding</a>  <a href="https://github.com/genomicsclass/labs/blob/master/batch/confounding.Rmd">[Rmd]</a></li>
  <li><a href="pages/confounding_exercises.html">Confounding exercises</a></li>
  <li><a href="pages/eda_with_pca.html">EDA with PCA</a>  <a href="https://github.com/genomicsclass/labs/blob/master/batch/eda_with_pca.Rmd">[Rmd]</a></li>
  <li><a href="pages/eda_with_pca_exercises.html">EDA with PCA exercises</a></li>
  <li><a href="pages/adjusting_with_linear_models.html">Adjusting with linear models</a>  <a href="https://github.com/genomicsclass/labs/blob/master/batch/adjusting_with_linear_models.Rmd">[Rmd]</a></li>
  <li><a href="pages/adjusting_with_linear_models_exercises.html">Adjusting with linear models exercises</a></li>
  <li><a href="pages/factor_analysis.html">Factor analysis</a>  <a href="https://github.com/genomicsclass/labs/blob/master/batch/factor_analysis.Rmd">[Rmd]</a></li>
  <li><a href="pages/factor_analysis_exercises.html">Factor analysis exercises</a></li>
  <li><a href="pages/adjusting_with_factor_analysis.html">Adjusting with factor analysis</a>  <a href="https://github.com/genomicsclass/labs/blob/master/batch/adjusting_with_factor_analysis.Rmd">[Rmd]</a></li>
  <li><a href="pages/adjusting_with_factor_analysis_exercises.html">Adjusting with factor analysis exercises</a></li>
</ul>

<hr />

<p><a name="5xtop"></a></p>

<h2 id="chapter-11---introduction-to-bioconductor">Chapter 11 - Introduction to Bioconductor</h2>

<ul>
  <li><a href="https://github.com/mikelove/bioc-refcard">Mike Love’s general reference card</a></li>
  <li><a href="pages/biomotiv.html">Motivations and core values (optional)</a></li>
  <li><a href="pages/installing_Bioconductor_finding_help.html">Installing Bioconductor and finding help</a> <a href="https://github.com/genomicsclass/labs/blob/master/bioc/installing_Bioconductor_finding_help.Rmd">[Rmd]</a></li>
  <li><a href="pages/dataman2019.html">Data structure and management for genome scale experiments</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocintro_5x/dataman2019.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/dataman2019.html#coord">Coordinating multiple tables: ExpressionSet</a></li>
      <li><a href="pages/dataman2019.html#geoquery">Institutional archives: GEO, ArrayExpress</a></li>
    </ul>
  </li>
  <li><a href="pages/bioc1_igranges.html">Interlude: Working with general genomic features using GenomicRanges</a>
    <ul>
      <li><a href="pages/bioc1_igranges.html#IRanges">IRanges introduced</a></li>
      <li><a href="pages/bioc1_igranges.html#intrarange">Intra-range operations</a></li>
      <li><a href="pages/bioc1_igranges.html#interrange">Inter-range operations</a></li>
      <li><a href="pages/bioc1_igranges.html#granges">GRanges</a></li>
      <li><a href="pages/bioc1_igranges.html#findoverlaps">Calculating overlaps</a></li>
    </ul>
  </li>
  <li><a href="pages/dataman2019.html#summexp">Range-oriented solutions for current experimental paradigms</a>
    <ul>
      <li><a href="pages/dataman2019.html#summexp">SummarizedExperiment: for RNA-seq and 450k methylation</a></li>
      <li><a href="pages/dataman2019.html#extern">External storage for very large assays</a></li>
      <li><a href="pages/dataman2019.html#genomicFiles">GenomicFiles for families of BAM or BED</a></li>
      <li><a href="pages/dataman2019.html#variants">DNA Variants: VCF handling with VariantAnnotation and VariantTools</a></li>
      <li><a href="pages/dataman2019.html#multiomic">Handling multiomic archives like TCGA</a></li>
      <li><a href="pages/dataman2019.html#cloud">Cloud-oriented solutions: e.g., Google BigQuery</a></li>
    </ul>
  </li>
  <li><a href="pages/bioc1_align.html">Short read mapping/alignment software (optional)</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_align.Rmd">[Rmd]</a></li>
</ul>

<h2 id="chapter-12---genomic-annotation-with-bioconductor">Chapter 12 - Genomic Annotation with Bioconductor</h2>

<ul>
  <li><a href="pages/bioc1_igranges.html">More details on GRanges</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_igranges.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc1_igranges.html#rleviews">Run-length encoding, views</a></li>
      <li><a href="pages/bioc1_igranges.html#genomic">Application to genomic landmarks</a></li>
      <li><a href="pages/bioc1_igranges.html#methyviz">Application to 450k methylation array visualization</a></li>
    </ul>
  </li>
  <li><a href="pages/bioc1_annoOverview.html">General overview of Bioconductor annotation</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_annoOverview.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc1_annoOverview.html#threelev">Levels: reference sequence, regions of interest, pathways</a></li>
      <li><a href="pages/bioc1_annoOverview.html#findingref">Discovering reference sequence</a></li>
      <li><a href="pages/bioc1_annoOverview.html#hsap">A build of the human genome</a></li>
      <li><a href="pages/bioc1_annoOverview.html#txUCSCnENSEMBLE">Gene/Transcript/Exon catalogs from UCSC and Ensembl</a></li>
      <li><a href="pages/bioc1_annoOverview.html#importExport">Importing and exporting regions and scores</a></li>
      <li><a href="pages/bioc1_annoOverview.html#annhub">AnnotationHub: brokering thousands of annotation resources</a></li>
      <li><a href="pages/bioc1_annoOverview.html#orgdb">OrgDb: simple interface to annotation databases</a></li>
      <li><a href="pages/bioc1_annoOverview.html#genesets">Finding and managing gene sets</a></li>
      <li><a href="pages/bioc1_annoOverview.html#organismdb">OrganismDb: unifying diverse annotation</a></li>
    </ul>
  </li>
  <li><a href="pages/bioc1_annoCheat.html">Cheat sheet on Bioconductor annotation</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_annoCheat.Rmd">[Rmd]</a></li>
  <li><a href="pages/bioc1_liftOver.html">Translating addresses between genome builds: liftOver</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_liftOver.Rmd">[Rmd]</a></li>
</ul>

<h2 id="chapter-13---genome-scale-hypothesis-testing-with-bioconductor">Chapter 13 - Genome-scale hypothesis testing with Bioconductor</h2>

<ul>
  <li><a href="pages/bioc1_btvari.html">Distinguishing biological and technical variability</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_btvari.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc1_btvari.html#pooling">An experiment with pooled and individual samples</a></li>
      <li><a href="pages/bioc1_btvari.html#techContrast">Measuring technical variation</a></li>
      <li><a href="pages/bioc1_btvari.html#biovar">Measuring biological variation</a></li>
      <li><a href="pages/bioc1_btvari.html#interpretation">Interpretation</a></li>
    </ul>
  </li>
  <li><a href="pages/bioc1_t_mult.html">Multiple comparisons with genewise t-tests</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_t_mult.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc1_t_mult.html#rowWiseT">Gene-wise testing</a></li>
      <li><a href="pages/bioc1_t_mult.html#naive">Naive enumeration of genes</a></li>
      <li><a href="pages/bioc1_t_mult.html#sham">Demonstrating danger of multiple testing with a set of sham comparisons</a></li>
      <li><a href="pages/bioc1_t_mult.html#adjusted">Adjusting for multiplicity with qvalue</a></li>
      <li><a href="pages/bioc1_t_mult.html#demo">Adjusted counts in the sham case</a></li>
    </ul>
  </li>
  <li><a href="pages/bioc1_limma.html">Moderated t tests via limma</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_limma.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc1_limma.html#spikein">A spike-in dataset</a></li>
      <li><a href="pages/bioc1_limma.html#simpleT">Naive t-tests</a></li>
      <li><a href="pages/bioc1_limma.html#limmaSteps">Three steps with limma: lmFit, eBayes, topTable</a></li>
      <li><a href="pages/bioc1_limma.html#expose">Exposing the spiked-in genes</a></li>
      <li><a href="pages/bioc1_limma.html#shrinkage">A view of the shrinkage of variance estimates</a></li>
    </ul>
  </li>
  <li><a href="pages/bioc1_geneset_1.html">Introducing gene sets and gene set analysis</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_geneset_1.Rmd">[Rmd]</a>
    <ul>
      <li>Data wrangling
        <ul>
          <li><a href="pages/bioc1_geneset_1.html#dataprep">A dataset for comparing expression by gender</a></li>
          <li><a href="pages/bioc1_geneset_1.html#batch">Finding surrogate variables/batch effect correction</a></li>
        </ul>
      </li>
      <li><a href="pages/bioc1_geneset_1.html#MSIGDB">The Broad Institute MsigDb</a>
        <ul>
          <li><a href="pages/bioc1_geneset_1.html#identifierMapping">Identifier remapping</a></li>
          <li><a href="pages/bioc1_geneset_1.html#chisq">Categorical testing</a></li>
          <li><a href="pages/bioc1_geneset_1.html#wilcox">Statistical summaries for sets: Wilcoxon</a></li>
          <li><a href="pages/bioc1_geneset_1.html#ttest">Statistical summaries for sets: t statistics</a></li>
        </ul>
      </li>
      <li><a href="pages/bioc1_geneset_1.html#correl">Adjusting for within-set correlation</a></li>
      <li><a href="pages/bioc1_geneset_1.html#permutations">A permutation procedure</a></li>
    </ul>
  </li>
</ul>

<h2 id="chapter-14---visualization-of-genome-scale-data">Chapter 14 - Visualization of genome scale data</h2>

<ul>
  <li><a href="pages/bioc2_vizOv.html">A basic overview of visualization tasks and strategies</a><a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_vizOv.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc2_vizOv.html#genemods">Gene models</a></li>
      <li><a href="pages/bioc2_vizOv.html#genemodsData">Gene models plus data</a></li>
      <li><a href="pages/bioc2_vizOv.html#vizFunc">Driving visualizations with functions</a></li>
      <li><a href="pages/bioc2_vizOv.html#vizShiny">Using the browser to drive visualization functions via shiny</a></li>
      <li><a href="pages/bioc2_vizOv.html#vizPlotly">Queriable dynamic displays with plotly</a></li>
    </ul>
  </li>
  <li>Annotation-oriented visualizations
    <ul>
      <li><a href="pages/bioc2_ggbio.html">Sketching the binding landscape over chromosomes with ggbio’s karyogram layout</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_ggbio.Rmd">[Rmd]</a></li>
      <li><a href="pages/bioc2_gvfeat.html">Plotting data in the context of genomic features with Gviz</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_gvfeat.Rmd">[Rmd]</a></li>
    </ul>
  </li>
  <li><a href="pages/bioc2_vizNGS.html">Visualizing NGS data</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_vizNGS.Rmd">[Rmd]</a></li>
  <li>Interactive visualization
    <ul>
      <li><a href="pages/bioc2_shiny.html">Graphical user interfaces for multivariate data with shiny</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_shiny.Rmd">[Rmd]</a></li>
      <li><a href="pages/esHclust.html">Clustering gene expression data with shiny</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/esHclust.Rmd">[Rmd]</a></li>
    </ul>
  </li>
  <li><a href="pages/finalViz.html">Final remarks on visualization</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/finalViz.Rmd">[Rmd]</a></li>
</ul>

<h2 id="chapter-15-pursuing-scalability-in-genomic-analysis-parallelism-and-out-of-memory-data">Chapter 15: Pursuing scalability in genomic analysis: parallelism and out-of-memory data</h2>

<ul>
  <li><a href="pages/bioc2_parallel.html">Parallel computing with R and Bioconductor</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_parallel.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc2_parallel.html#speedup">Demonstrating simple speedup in multicore environments</a></li>
      <li><a href="pages/bioc2_parallel.html#implicit">Implicit parallelism with BiocParallel and GenomicAlignments</a></li>
    </ul>
  </li>
  <li><a href="pages/bioc2_externData.html">External data: data interfaces that spare RAM</a><a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_externData.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc2_externData.html#sqlite">SQLite for annotation</a></li>
      <li><a href="pages/bioc2_externData.html#tabix">Tabix-indexed BAM</a></li>
      <li><a href="pages/bioc2_externData.html#hdf5">HDF5</a></li>
      <li><a href="pages/bioc2_nosql.html">An illustration of NoSQL with S4: mongodb and RaggedMongoExpt</a><a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_nosql.Rmd">[Rmd]</a></li>
    </ul>
  </li>
  <li><a href="pages/multiOOM.html">Benchmarking various out-of-memory solutions</a><a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/multiOOM.Rmd">[Rmd]</a></li>
  <li><a href="pages/bioc2_HPCami.html">Introduction to Bioconductor’s Amazon Machine Instance for cluster creation and use in EC2</a> <a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_HPCami.Rmd">[Rmd]</a></li>
  <li><a href="pages/bioc2_hybstor.html">Sharded GRanges for scalable integrative analysis</a><a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_hybstor.Rmd">[Rmd]</a></li>
</ul>

<h3 id="chapter-16-multi-omic-data-integration">Chapter 16: Multi-omic data integration</h3>

<ul>
  <li><a href="pages/bioc2_integExamps.html">Basic examples of multi-omic integration</a><a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_integExamps.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc2_integExamps.html#yeasttfs">Transcription factor (TF) binding and gene coexpression in yeast</a></li>
      <li><a href="pages/bioc2_integExamps.html#gwastf">TF binding and GWAS hits in humans</a></li>
    </ul>
  </li>
  <li><a href="pages/tcga.html">Using RTCGAToolbox outputs to integrate clinical, mutation, expression and methylation assays</a><a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/tcga.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/tcga.html#firehose">Basic data acquisition</a></li>
      <li><a href="pages/tcga.html#clinical">Working with clinical data</a>
        <ul>
          <li><a href="pages/tcga.html#severity">Defining a severity marker</a></li>
          <li><a href="pages/tcga.html#survival">Extracting survival times</a></li>
        </ul>
      </li>
      <li><a href="pages/tcga.html#mutation">Working with mutations</a></li>
      <li><a href="pages/tcga.html#curation">Curation tasks for discrepant identifier formats</a></li>
      <li><a href="pages/tcga.html#expression">Working with expression data</a>
        <ul>
          <li><a href="pages/tcga.html#tstageToExpr">Associating tumor stage with expression patterns</a></li>
          <li><a href="pages/tcga.html#methy">Linking DNA methylation with expression patterns</a></li>
        </ul>
      </li>
    </ul>
  </li>
  <li><a href="pages/bioc2_rainfall.html">Application to visualization: kataegis and rainfall plot</a><a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_rainfall.Rmd">[Rmd]</a></li>
</ul>

<h3 id="chapter-17-fostering-reproducible-genome-scale-analysis">Chapter 17: Fostering reproducible genome-scale analysis</h3>

<ul>
  <li><a href="pages/bioc2_repro1.html">Overview of unit on reproducibility</a><a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_repro1.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc2_repro1.html#defs">Basic definitions</a></li>
      <li><a href="pages/bioc2_repro1.html#infrastr">Infrastructure requirements</a></li>
      <li><a href="pages/bioc2_repro1.html#infrastr">Statistical aspects of reproducibility</a></li>
      <li><a href="pages/bioc2_repro1.html#RP">Analysis of reproducibility probability (Boos and Stefanski 2011)</a></li>
      <li><a href="pages/bioc2_repro1.html#costs">Costs of highly reproducible designs</a></li>
    </ul>
  </li>
  <li><a href="pages/bioc2_rpacks.html">Package structure, creation, installation, management</a><a href="https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_rpacks.Rmd">[Rmd]</a>
    <ul>
      <li><a href="pages/bioc2_rpacks.html#defpack">What is a package?</a></li>
      <li><a href="pages/bioc2_rpacks.html#skel">Using <code class="language-plaintext highlighter-rouge">package.skeleton</code></a></li>
      <li><a href="pages/bioc2_rpacks.html#org">Using <code class="language-plaintext highlighter-rouge">makeOrganismPackage</code></a></li>
      <li><a href="pages/bioc2_rpacks.html#devtools">Using <code class="language-plaintext highlighter-rouge">devtools</code></a>
        <ul>
          <li><a href="pages/bioc2_rpacks.html#create"><code class="language-plaintext highlighter-rouge">create()</code> to set up folders and DESCRIPTION</a></li>
          <li><a href="pages/bioc2_rpacks.html#source">Composing documentation plus code</a></li>
          <li><a href="pages/bioc2_rpacks.html#docinst"><code class="language-plaintext highlighter-rouge">document()</code>, <code class="language-plaintext highlighter-rouge">install()</code></a></li>
        </ul>
      </li>
      <li><a href="pages/bioc2_rpacks.html#wrapup">Conclusions, including a link to a recent Nature Toolbox article on Bioconductor</a></li>
    </ul>
  </li>
</ul>

<hr />

<h2 id="legacy-material-from-2015-introduction-to-bioconductor">Legacy material from 2015 Introduction to Bioconductor</h2>

<ul>
  <li><a href="pages/installing_Bioconductor_finding_help.html">Installing Bioconductor and finding help</a></li>
  <li><a href="pages/annoPhen.html">Annotating phenotypes and molecular function</a></li>
  <li>
    <p><a href="pages/eset.html">The ExpressionSet Container</a></p>
  </li>
  <li><a href="pages/iranges_granges.html">IRanges and GRanges</a></li>
  <li>
    <p><a href="pages/operateGRanges.html">Operating on GRanges</a></p>
  </li>
  <li><a href="pages/annoCheat.html">Cheat sheet for genomic annotation</a></li>
  <li><a href="pages/anno4liftover.html">Translating addresses between reference builds with liftOver</a></li>
  <li>
    <p><a href="https://github.com/mikelove/bioc-refcard/blob/master/README.Rmd">Cheat sheet for GRanges and other Bioconductor objects</a></p>
  </li>
  <li><a href="pages/import_NGS.html">Importing NGS data with Bioconductor</a></li>
  <li>
    <p><a href="pages/read_counting.html">NGS read counting</a></p>
  </li>
  <li><a href="pages/biological_versus_technical_var.html">Technical versus biological variability</a></li>
  <li><a href="pages/inference_with_bioc.html">Statistical Inference with Bioconductor</a></li>
  <li><a href="pages/using_limma.html">Using limma</a></li>
  <li><a href="pages/gene_set_analysis.html">Gene Sets Analysis</a></li>
  <li><a href="pages/gene_set_analysis_in_R.html">Gene Sets Analysis in R</a></li>
</ul>

<!--

new ones:
- [Computing with chromosomes and variants](pages/chromComp.html)

old ones:
- [Reading microarray data](pages/reading_microarray_data.html)
- [Downloading data from GEO using GEOquery](pages/GEOquery.html)
- [Background](pages/background.html)
- [Normalization](pages/normalization.html)
- [SVA, ComBat, PCA and SVD](pages/svacombat.html)
- [Basic inference for microarray](pages/basic_inference_microarray.html)
- [EDA plots for microarray](pages/EDA_plots_for_microarray.html)
- [EDA plots for next generation sequencing](pages/EDA_plots_for_NGS.html)
- [Visualizing NGS data](pages/visualizing_NGS.html)
- [Mapping features to genes](pages/mapping_features.html)
- [Gene set analysis lecture](pages/gsa.html)
- [Gene set testing in R](pages/gene_set_testing.html)

-->

<h2 id="rna-seq-data-analysis">RNA-seq data analysis</h2>

<ul>
  <li><a href="https://github.com/genomicsclass/labs/blob/master/rnaseq/fastq.md">Downloading and unzipping fastq files</a></li>
  <li><a href="https://github.com/genomicsclass/labs/blob/master/rnaseq/genome_align_STAR.md">Genomic alignment with STAR</a></li>
  <li>
    <p><a href="https://github.com/genomicsclass/labs/blob/master/rnaseq/trancsript_align_RSEM.md">Transcriptome alignment with RSEM</a></p>
  </li>
  <li><a href="pages/rnaseq_gene_level.html">RNA-seq at the gene-level: EDA, DE and SVA</a></li>
  <li><a href="pages/rnaseq_exon_usage.html">Differential exon usage</a></li>
  <li><a href="pages/rnaseq_isoform_cummerbund.html">Exploring plots of isoform-level expression with Cufflinks/cummeRbund</a></li>
</ul>

<h2 id="variant-discovery-and-genotyping">Variant Discovery and Genotyping</h2>

<ul>
  <li><a href="pages/SNP.html">Genome variation</a> from 2014</li>
</ul>

<h2 id="chip-seq-data-analysis">ChIP-seq data analysis</h2>

<ul>
  <li><a href="pages/ChIPseq.html">ChIP-seq</a> from 2014</li>
</ul>

<h2 id="dna-methylation-data-analysis">DNA methylation data analysis</h2>

<ul>
  <li><a href="pages/methylation.html">DNA Methylation Data Analysis</a></li>
  <li><a href="pages/minfi.html">Reading 450K idat files with the minfi package</a></li>
  <li>
    <p><a href="pages/epiviz.html">Interactive visualization of DNA methylation data analysis</a></p>
  </li>
  <li><a href="pages/inference_for_DNAmeth.html">Statistical Inference in the Analysis of DNA methylation Data</a></li>
</ul>

<hr />

<h2 id="footnotes-for-all-lectures">Footnotes for all lectures</h2>

<ul>
  <li><a href="pages/footnotes.html">Footnotes</a></li>
</ul>

<h2 id="acknowledgments">Acknowledgments</h2>

<ul>
  <li><a href="pages/acknowledgments.html">Course acknowledgments</a></li>
</ul>


  </div>
</div>


      </div>
      <hr>
      <footer>
        <p><small>
  <!-- start of Karl's footer; modify this part -->

<a href="http://genomicsclass.github.io/book/">PH525x</a>, 
Rafael Irizarry and Michael Love, 
<a href="https://github.com/genomicsclass/labs/blob/master/LICENSE">MIT License</a>

  <!-- end of Karl's footer; modify this part -->
        </small></p>
      </footer>

    </div>

    
  </body>
</html>

